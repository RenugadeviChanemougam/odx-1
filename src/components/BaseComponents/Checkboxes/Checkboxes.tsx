import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import FormGroup, { makeItemId } from '../FormGroup/FormGroup';

// TODO - fix data handling
export default function Checkboxes(props) {
  const {
    name,
    label,
    items,
    isSmall,
    labelIsHeading,
    errorText,
    hintText,
    required,
    onChange,
    onBlur,
    inputProps
  } = props;

  const checkboxClasses = `govuk-checkboxes ${isSmall ? 'govuk-checkboxes--small' : ''}`;
  const itemClasses = 'govuk-checkboxes__item'
  const checkboxItemClasses = 'govuk-checkboxes__input'
  const hintTextClasses = `govuk-hint govuk-checkboxes__hint`
  const labelClasses = `govuk-label govuk-checkboxes__label`

  // const initialiseChecked = () => {
  //   const values = []
  //   items.forEach(item => values.push(item.label))
  //   return values
  // }
  // const [values, setValues] = useState([])


  return (
    <FormGroup {...props}>
      <div className={checkboxClasses}>
        {/* TODO - add item ids and names */}
        {items.map((item, index) => (
          <div className={itemClasses} key={makeItemId(index, name)}>
            <input
              className={checkboxItemClasses}
              {...inputProps}
              id={makeItemId(index, name)}
              name={makeItemId(index, name)}
              type='checkbox'
              value={item.checked}
              onChange={!item.readOnly ? onChange : ()=>{}}
              onBlur={!item.readOnly ? onBlur : ()=>{}}
            ></input>
            <label className={labelClasses}>{item.label}</label>
            {item.hintText ? <div id={makeItemId(index, `${name}-item-hint`)} className={hintTextClasses}>{item.hintText}</div>: null}
          </div>
        ))}
      </div>
    </FormGroup>
  );
}

Checkboxes.propTypes = {
  ...FormGroup.propTypes,
  name: PropTypes.string,
  label: PropTypes.string,
  isSmall: PropTypes.bool,
  labelIsHeading: PropTypes.bool,
  items: PropTypes.array,
  required: PropTypes.bool,
  errorText: PropTypes.string,
  inputProps: PropTypes.object,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
};
