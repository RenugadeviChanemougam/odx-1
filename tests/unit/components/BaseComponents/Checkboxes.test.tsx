import React from 'react';
import renderer from 'react-test-renderer';
import Checkboxes from '../../../../src/components/BaseComponents/Checkboxes/Checkboxes';

let Props = Checkboxes.propTypes;



it("renders a group of 3 checkboxes, with legend 'heading' ", () => {
  let items = [
    { checked: false, label: 'Item 1', hintText: '', readOnly: false },
    { checked: false, label: 'Item 2', hintText: '', readOnly: false },
    { checked: false, label: 'Item 3', hintText: '', readOnly: false }
  ];
  const tree = renderer.create(<Checkboxes name="test-checkbox" label="heading" items={items} isSmall={false} labelIsHeading={true} errorText="" />).toJSON();
  expect(tree).toMatchSnapshot()
});


it("renders a group of 3 checkboxes, hint text for label", () => {
  let items = [
    { checked: false, label: 'Item 1', hintText: '', readOnly: false },
    { checked: false, label: 'Item 2', hintText: '', readOnly: false },
    { checked: false, label: 'Item 3', hintText: '', readOnly: false }
  ];
  const tree = renderer.create(<Checkboxes name="test-checkbox" label="heading" items={items} isSmall={false} labelIsHeading={true} errorText="" hintText={"Testing hint text"} />).toJSON();
  expect(tree).toMatchSnapshot()
});

it("renders a group of 3 checkboxes, hint text for items", () => {
  let items = [
    { checked: false, label: 'Item 1', hintText: 'testing item hint text', readOnly: false },
    { checked: false, label: 'Item 2', hintText: '', readOnly: false },
    { checked: false, label: 'Item 3', hintText: '', readOnly: false }
  ];
  const tree = renderer.create(<Checkboxes name="test-checkbox" label="heading" items={items} isSmall={false} labelIsHeading={true} errorText="" />).toJSON();
  expect(tree).toMatchSnapshot()
});


// TODO - Add this test case after Fieldset.tsx has been merged as it has the logic for error

// it("renders a group of 3 checkboxes, with error as expected ", () => {
//   let items = [
//     { checked: false, label: 'Item 1', hintText: '', readOnly: false },
//     { checked: false, label: 'Item 2', hintText: '', readOnly: false },
//     { checked: false, label: 'Item 3', hintText: '', readOnly: false }
//   ];
//   const tree = renderer.create(<Checkboxes label="heading" items={items} isSmall={false} labelIsHeading={true} errorText="A testing Error." />).toJSON();
//   expect(tree).toMatchSnapshot()
// });


// TODO - Add Test cases for functionality
