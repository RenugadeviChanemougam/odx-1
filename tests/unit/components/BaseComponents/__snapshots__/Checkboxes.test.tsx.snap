// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a group of 3 checkboxes, hint text for items 1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="test-checkbox"
    >
      heading
    </label>
  </h1>
  <div
    className="govuk-checkboxes "
  >
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox"
        name="test-checkbox"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 1
      </label>
      <div
        className="govuk-hint govuk-checkboxes__hint"
        id="test-checkbox-item-hint"
      >
        testing item hint text
      </div>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-1"
        name="test-checkbox-1"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 2
      </label>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-2"
        name="test-checkbox-2"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 3
      </label>
    </div>
  </div>
</div>
`;

exports[`renders a group of 3 checkboxes, hint text for label 1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="test-checkbox"
    >
      heading
    </label>
  </h1>
  <div
    className="govuk-hint"
    id="test-checkbox-hint"
  >
    Testing hint text
  </div>
  <div
    className="govuk-checkboxes "
  >
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox"
        name="test-checkbox"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 1
      </label>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-1"
        name="test-checkbox-1"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 2
      </label>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-2"
        name="test-checkbox-2"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 3
      </label>
    </div>
  </div>
</div>
`;

exports[`renders a group of 3 checkboxes, with legend 'heading'  1`] = `
<div
  className="govuk-form-group"
>
  <h1
    className="govuk-label-wrapper"
  >
    <label
      className="govuk-label govuk-label--l"
      htmlFor="test-checkbox"
    >
      heading
    </label>
  </h1>
  <div
    className="govuk-checkboxes "
  >
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox"
        name="test-checkbox"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 1
      </label>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-1"
        name="test-checkbox-1"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 2
      </label>
    </div>
    <div
      className="govuk-checkboxes__item"
    >
      <input
        className="govuk-checkboxes__input"
        id="test-checkbox-2"
        name="test-checkbox-2"
        type="checkbox"
        value={false}
      />
      <label
        className="govuk-label govuk-checkboxes__label"
      >
        Item 3
      </label>
    </div>
  </div>
</div>
`;
